executor/delete.go

```diff
@@ -132,50 +114,20 @@ func (e *DeleteExec) deleteSingleTableByChunk(ctx context.Context) error {
 	return nil
 }
 
-func (e *DeleteExec) initialMultiTableTblMap() {
-	e.tblMap = make(map[int64][]*ast.TableName, len(e.Tables))
-	for _, t := range e.Tables {
-		e.tblMap[t.TableInfo.ID] = append(e.tblMap[t.TableInfo.ID], t)
-	}
-}
-
-func (e *DeleteExec) getColPosInfos(schema *expression.Schema) []tblColPosInfo {
-	var colPosInfos []tblColPosInfo
-	// Extract the columns' position information of this table in the delete's schema, together with the table id
-	// and its handle's position in the schema.
-	for id, cols := range schema.TblID2Handle {
-		tbl := e.tblID2Table[id]
-		for _, col := range cols {
-			if !e.matchingDeletingTable(id, col) {
-				continue
-			}
-			offset := getTableOffset(schema, col)
-			end := offset + len(tbl.Cols())
-			colPosInfos = append(colPosInfos, tblColPosInfo{tblID: id, colBeginIndex: offset, colEndIndex: end, handleIndex: col.Index})
-		}
-	}
-	return colPosInfos
-}
-
-func (e *DeleteExec) composeTblRowMap(tblRowMap tableRowMapType, colPosInfos []tblColPosInfo, joinedRow []types.Datum) {
+func (e *DeleteExec) composeTblRowMap(tblRowMap tableRowMapType, colPosInfos []plannercore.TblColPosInfo, joinedRow []types.Datum) {
 	// iterate all the joined tables, and got the copresonding rows in joinedRow.
 	for _, info := range colPosInfos {
-		if tblRowMap[info.tblID] == nil {
-			tblRowMap[info.tblID] = make(map[int64][]types.Datum)
+		if tblRowMap[info.TblID] == nil {
+			tblRowMap[info.TblID] = make(map[int64][]types.Datum)
 		}
-		handle := joinedRow[info.handleIndex].GetInt64()
-		// tblRowMap[info.tblID][handle] hold the row datas binding to this table and this handle.
-		tblRowMap[info.tblID][handle] = joinedRow[info.colBeginIndex:info.colEndIndex]
+		handle := joinedRow[info.HandleOrdinal].GetInt64()
+		// tblRowMap[info.TblID][handle] hold the row datas binding to this table and this handle.
+		tblRowMap[info.TblID][handle] = joinedRow[info.Start:info.End]
 	}
 }
 
 func (e *DeleteExec) deleteMultiTablesByChunk(ctx context.Context) error {
-	if len(e.Tables) == 0 {
```

Comment:
I checked the grammer in parser. If it's a multi delete. The `Tables` cannot be nil.

Reply:
OK